_posts/2020-01-12-hhvm-4.92.markdown

```diff
@@ -0,0 +1,34 @@
+---
+title: "HHVM 4.92"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.92 is released! This release marks the end of support for 4.83;
+HHVM 4.84&ndash;4.91 remain supported, as do the 4.56 and 4.80 LTS releases.
+
+(Note: HHVM 4.85, 4.89 and 4.90 were skipped.)
+
+# Highlights
+
+- If a class A is a supertype of class/interface B via `require extends`, it
+  is now able to call its' own private methods on instances of B.
+- Corrected error line number for reified generic errors.
+- Replaced sometimes-confusing "expected dynamic, got nonnull" with clearer
```

Comment:
is it "got ?nonnull" (equivalent to mixed) or "got nonnull" (not equivalent)?

Reply:
This is for situations where you have a mixed, check it's not-null, then try to use it as dynamic.

`nonnull` is technically more correct, but users seem to to generally still think of their values as `mixed` instead - or, they read this as implying that the value should be null.